seqr/views/utils/dataset_utils.py

```diff
@@ -374,17 +386,22 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
             if indiv_id and sample_id not in sample_id_to_individual_id_mapping:
                 sample_id_to_individual_id_mapping[sample_id] = indiv_id
 
+            project = row_dict.pop(PROJECT_COL, None)
+            if project:
+                sample_id_project_tuples.add((sample_id, project))
             samples_by_id[sample_id][gene_id] = row_dict
 
     message = f'Parsed {len(samples_by_id)} RNA-seq samples'
     info = [message]
     logger.info(message, user)
 
     data_source = file_path.split('/')[-1].split('_-_')[-1]
+    projects = Project.objects.filter(name__in={project for _, project in sample_id_project_tuples})
     samples, remaining_sample_ids = _match_and_update_rna_samples(
-        projects=get_internal_projects(),
+        projects=projects if len(projects) > 0 else get_internal_projects(),
```

Comment:
This is too fragile/confusing. The project filtration is different for outlier and tpm models - that means you need to explicitly pass different projects in from those 2 methods

Reply:
I could pass in `projects` for outliers, but the `projects` for the TPMs will be available after the data is read from the file. It's more consistent to get the projects here. What do you think about that?